landing_page/app/controllers/microposts_controller.rb

```diff
@@ -38,7 +38,11 @@ def recover
     helper_method :recover
 
     def index 
-        @microposts = Micropost.kept.order(created_at: :desc) 
+        if params[:archived] == "true"
+            @microposts = Micropost.kept.where("created_at < ?", Time.now-7.days).order(created_at: :desc)
+        else
+            @microposts = Micropost.kept.where("created_at > ?", Time.now-7.days).order(created_at: :desc)
+        end       
```

Comment:
There's an opportunity for improvement in readability here.

At first it's hard to notice what's the difference between the two lines, that's code duplication. They differ by only a single character.